### DISCUSSION TITLE
How can I get the total following count of a user?

### DISCUSSION THREAD

**Discussion by emreond** [sbertix/Swiftagram#175] [Q&A] [answered]
Hello,

I want to get the total following count of a user. I am using `Endpoint.User.summary(for: identifier)` for getting following count, thumbnail and userName. However, this endpoint returns nill for `counter?.following`. What is the proper way of getting these informations?

**P.S:** What I want to achieve is that I want to create a progress bar for fetching all followings so It is also ok for me to get the total following count in `friendShip.following` endpoint If possible.

**Comment by sbertix** [comment, marked as answer]
Hey @emreond, 

If `counter` is `nil` it means the data you're looking for is not populated in the response. 

It shouldn't be `nil` when calling `Endpoint.User.summary(for:)`, though.
Do you mind sending me the console log for the response?
Assuming you're calling it like this, add the logging call.

```swift
Endpoint.User.summary(for: identifier)
   .unlocking(with: secret)
   .task { _ in }
   .logging(level: .responseBody)
   .resume()
```

**Reply by emreond** [reply]
:( My bad I was using user object from search endpoint in completion block not the one got from summary endpoint  (with one misspelling character :D). Sorry about that.

**Reply by sbertix** [reply]
No worries 😊 
I'm glad it worked out.

**Reply by emreond** [reply]
Also may I ask If identifier variable is unique in User object?  I am using realm database and I use this variable as primary key for users. It was `primaryKey` in previous library but in this one I changed it to `identifier`

**Reply by sbertix** [reply]
It is 😊 it's exactly the primary key.
